Isosurfacing_3/include/CGAL/Isosurfacing_3/interpolation_schemes_3.h

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2022-2024 INRIA Sophia-Antipolis (France), GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Mael Rouxel-Labbé
+
+#ifndef CGAL_ISOSURFACING_3_INTERNAL_INTERPOLATION_SCHEMES_3_H
+#define CGAL_ISOSURFACING_3_INTERNAL_INTERPOLATION_SCHEMES_3_H
+
+#include <CGAL/license/Isosurfacing_3.h>
+
+#include <CGAL/enum.h>
+
+#include <array>
+#include <vector>
+
+namespace CGAL {
+namespace Isosurfacing {
+
+/*!
+ * \ingroup IS_Fields_helpers_grp
+ *
+ * \cgalModels{IsosurfacingInterpolationScheme_3}
+ *
+ * The class `Trilinear_interpolation` is the standard interpolation scheme
+ * to define continuous fields of scalar values and gradients from data defined
+ * only at the vertices of a %Cartesian grid.
+ *
+ * \tparam Grid must be `CGAL::Isosurfacing::Cartesian_grid_3<GeomTraits>`, with `GeomTraits`
+ *              a model of `IsosurfacingTraits_3`
+ */
+template <typename Grid>
+class Trilinear_interpolation
+{
+public:
+  using Geom_traits = typename Grid::Geom_traits;
+  using FT = typename Geom_traits::FT;
+  using Point_3 = typename Geom_traits::Point_3;
+  using Vector_3 = typename Geom_traits::Vector_3;
+  using Iso_cuboid_3 = typename Geom_traits::Iso_cuboid_3;
+
+public:
+  /*!
+   * \brief interpolates the values at a given point using trilinear interpolation.
+   *
+   * \param p the point at which to interpolate the values
```

Comment:
"at which to" does not sound very well.
"At which values are interpolated"? (same below)